File: patterns/design-patterns/src/main/java/com/baeldung/hexagonal/FileWriter.java
package com.baeldung.hexagonal;

import java.io.BufferedWriter;
import java.io.IOException;

public class FileWriter implements Writer {

    private static final String FILENAME = "E:\\test\\filename.txt";

    @Override
    public void write(String text) {
        try (BufferedWriter bw = new BufferedWriter(new java.io.FileWriter(FILENAME))) {
            bw.write(text);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: core-kotlin/src/main/kotlin/com/baeldung/filesystem/FileWriter.kt
package com.baeldung.filesystem

import java.io.File

class FileWriter {

    fun writeFileUsingPrintWriter(fileName: String, fileContent: String) =
            File(fileName).printWriter().use { out -> out.print(fileContent) }

    fun writeFileUsingBufferedWriter(fileName: String, fileContent: String) =
            File(fileName).bufferedWriter().use { out -> out.write(fileContent) }

    fun writeFileDirectly(fileName: String, fileContent: String) =
            File(fileName).writeText(fileContent)

    fun writeFileDirectlyAsBytes(fileName: String, fileContent: String) =
            File(fileName).writeBytes(fileContent.toByteArray())

}